src/main/java/lumina/task/TaskList.java

```diff
@@ -0,0 +1,332 @@
+package lumina.task;
+
+import lumina.exception.LuminaException;
+import lumina.parser.Parser;
+import lumina.ui.Ui;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.stream.Collectors;
+
+/**
+ * Manages a list of tasks and handles various operations such as adding,
+ * listing, and marking tasks as done or not done.
+ */
+public class TaskList {
+
+    // tasks
+    private ArrayList<Task> tasks;
+    private Ui ui;
+    private Parser parser;
+
+    /**
+     * Constructs a TaskList with the specified Ui and Parser.
+     *
+     * @param ui the Ui instance for displaying messages to the user
+     * @param parser the Parser instance for parsing inputs
+     */
+    public TaskList(Ui ui, Parser parser) {
+        this.ui = ui;
+        this.parser = parser;
+        this.tasks = new ArrayList<>();
+    }
+
+    /**
+     * Adds a task to the list and prints a confirmation message.
+     *
+     * @param task the Task to be added to the list
+     */
+    public void addTask(Task task) {
+        this.tasks.add(task);
+        StringBuilder addTaskMessage = new StringBuilder();
+        addTaskMessage.append("Got it. I've added this task:\n");
+        addTaskMessage.append(ui.indentMessage(task.toString())).append("\n");
+        addTaskMessage.append(String.format("Now you have %d tasks in the list.", tasks.size()));
+        ui.printMessage(addTaskMessage.toString());
+    }
+
+    /**
+     * Lists all tasks in the list and prints them.
+     */
+    public void listTasks() {
+        StringBuilder listedTaskMessage = new StringBuilder();
+        listedTaskMessage.append("Here are the tasks in your list:\n");
+        for (int i = 0; i < this.tasks.size(); i++) {
+            listedTaskMessage.append(Integer.toString(i + 1)).append(".");
+            listedTaskMessage.append(this.tasks.get(i));
+            if (i < this.tasks.size() - 1) {
+                listedTaskMessage.append("\n");
+            }
+        }
+        ui.printMessage(listedTaskMessage.toString());
```

Comment:
Instead of passing the message, would passing the tasks ArrayList to Ui to print be a better abstraction?